actions/steps/setup-custard/action.yaml

```diff
@@ -22,46 +22,87 @@ inputs:
   ci-setup:
     description: The CI setup for the package path.
     required: true
-  project-id:
-    description: The Google Cloud project ID.
-    required: false
-  workload-identity-provider:
-    description: The Google Cloud workload identity provider.
-    required: false
-  service-account:
-    description: The Google Cloud service account to use for credentials.
+  id-token:
+    description: The ID token to export variable, not exported if not provided.
     required: false
 
 runs:
+  # TODO: This should all be handled by the local runner.
+  #       When that's ready, we can call it directly and get rid of this.
   using: composite
   steps:
-    - name: Checkout
-      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
-    - name: Authenticate
-      uses: google-github-actions/auth@6fc4af4b145ae7821d527454aa9bd537d1f2dc5f # v2
-      if: inputs.project-id && inputs.workload-identity-provider && inputs.service-account
-      id: auth
-      with:
-        project_id: ${{ inputs.project-id }}
-        workload_identity_provider: ${{ inputs.workload-identity-provider }}
-        service_account: ${{ inputs.service-account }}
-        access_token_lifetime: 600s # 10 minutes
-        token_format: id_token
-        id_token_audience: https://action.test/ # service must have this custom audience
-        id_token_include_email: true
     - name: Export environment variables
       uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
       id: vars
       with:
         script: |
-          const { default: setupVars } = await import('${{ github.workspace }}/.github/scripts/setup-vars.js');
-          return await setupVars({
-            core,
-            projectId: '${{ inputs.project-id }}',
-            setup: ${{ inputs.ci-setup }},
-            serviceAccount: '${{ inputs.service-account }}',
-            idToken: '${{ steps.auth.outputs.id_token }}',
-          })
+          function uniqueId(length = 6) {
+            const min = 2 ** 32;
+            const max = 2 ** 64;
+            return Math.floor(Math.random() * max + min)
+              .toString(36)
+              .slice(0, length);
+          }
```

Comment:
This looks like we're embedding setup-vars.js directly in the step now?

Reply:
Yeah, the checkout on affected was not doing what we were expecting. It was checking out the caller repo, not the tools repo. So basically, the scripts file would have to exist in the language repo. This is just a temporary workaround, once we have the local runner this will no longer be necessary and will go away.

Reply:
Thanks for clarifying. Can you add these details to the PR description so we can more easily find the history later?

Reply:
I just added more details in the PR description